MatchAndPlan/src/transferCenterRoutePlanning/Util/RoutePlanByGreey.java

```diff
@@ -0,0 +1,570 @@
+package transferCenterRoutePlanning.Util;
+
+
+
+
+
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class RoutePlanByGreey {
+
+    private double[][] cityDistance;    //城市间距离
+    private List<LatLng> newPointsArray = null;
+    private List<LatLng> prePointsArray1;
+
+    public List<LatLng> mapToList(Map<String,LatLng> map)
+    {
+        ArrayList<LatLng> arrayList = new ArrayList<>();
+        for(Map.Entry<String,LatLng> entry : map.entrySet()){
+
+            arrayList.add(entry.getValue());
+        }
+        return arrayList;
+    }
+    //计算点间距离,并设置最小距离
+    public void getEachDistance(List<LatLng> allPoints) {
+        cityDistance = new double[allPoints.size()][allPoints.size()];     //设置数组大小
+        double min = 0;     //最短距离
+        //点到自身距离设为0
+        for(int k = 0;k < allPoints.size();k++) {
+            cityDistance[k][k] = 0;
+        }
+        //根据公式计算上三角
+        for (int i = 0;i < allPoints.size()-1;i++) {
+            for (int j = i+1; j < allPoints.size(); j++) {
+                double radLat1 = Rad(allPoints.get(i).latitude);
+                double radLat2 = Rad(allPoints.get(j).latitude);
+                double diffLat = Arith.sub(radLat1,radLat2);
+                double diffLng = Arith.sub(Rad(allPoints.get(i).longitude) , Rad(allPoints.get(j).longitude));
+                double distance = 2 * Math.asin(Math.sqrt(Math.pow(Math.sin(Arith.div(diffLat,2)), 2)
+                        + Math.cos(radLat1) * Math.cos(radLat2) * Math.pow(Math.sin(Arith.div(diffLng,2)), 2)));
+
+                distance = Arith.mul(distance,6378.137);     //地球半径
+                distance = Arith.div(Arith.round(Arith.mul(distance,10000),6),10000);
+                distance = Arith.div(Arith.round(Arith.mul(distance,1000),2),1);     //单位修改为米,取整
+
+                //计算最短距离
+                if(min>distance) {
+                    min = distance;
+                }
+
+                this.cityDistance[i][j] = Arith.round(distance,0);
+                this.cityDistance[j][i] = Arith.round(distance,0);
+            }
+
+        }
+    }
+
+    //经纬度转换成三角函数中度分表形式
+    private double Rad(double latOrLng) {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.